src/AssocArray.php

```diff
@@ -127,40 +163,68 @@ public function unique(?callable $comparator = null): AssocArray
             $result[$keyA] = $valueA;
         }
 
+        /** @phpstan-var array<TKey, TValue> $result */
         return new self($result);
     }
 
     /**
-     * @param mixed $value
+     * @phpstan-param TKey $key
+     * @phpstan-param TValue $value
+     * @phpstan-return AssocArray<TKey, TValue>
      */
-    public function with(string $key, $value): AssocArray
+    public function with($key, $value): AssocArray
     {
-        return $this->merge(new self([$key => $value]));
+        /** @phpstan-var array<TKey, TValue> $items */
+        $items = [$key => $value];
+
+        return $this->merge(new self($items));
     }
 
+    /**
+     * @phpstan-param AssocValue<TKey, TValue> $other
+     * @phpstan-return AssocArray<TKey, TValue>
+     */
     public function merge(AssocValue $other): AssocArray
     {
         return new self(array_merge($this->items, $other->toAssocArray()));
     }
 
-    public function without(string ...$keys): AssocArray
+    /**
+     * @phpstan-param TKey $key
+     * @phpstan-param array<int, TKey> $keys
+     * @phpstan-return AssocArray<TKey, TValue>
+     * @phpstan-ignore-next-line phpstan does not support generic variadics
+     */
+    public function without($key, ...$keys): AssocArray
```

Comment:
Why at least 1 key is required? 

I will require some additional logic in cases like:
```php
function applyConfig(AssocValue $assoc, Config $config): AssocValue {
    $keys = $config->exludedKeys();
    return $keys !== [] ? $assoc->without(...$config->exludedKeys()) : $assoc;
}
```